### PR TITLE
Single Screen Dev UI and container

### PR DESCRIPTION
**Pull request by Matt-Wehman**
1. Implemented single screen dev UI based on Figma mockups (no actual functionality like viewing other containers has been implemented)
2. Created a docker file to build a container for the UI
3. Updated each docker compose file to build the new container

### CODE REVIEW

**Comment by Matt-Wehman** [comment]
Small update:

Streaming the view of the vertical exhibit screens should now be functional.

This PR is ready for review

**Review by aaron97neu — COMMENTED** [review]
Few small things to fix up - the `docker-compose.yml` and `README.md` should likely just be reverted to their current state in main


In addition, I couldn't get anything to show up on the screens. Please confirm that this is intended as the placeholders mentioned in the PR?

image


I like the current UI. It's clean and reactive with snappy animations. Thanks.

File: README.md
[the remark this answers is not in the document]
Comment (Matt-Wehman) [reply README.md:1]: I'm not entirely sure how to maintain this either. For now, I just reverted to the old file manually.

File: docker-compose.yml
[the remark this answers is not in the document]
Comment (Matt-Wehman) [reply docker-compose.yml:1]: Updated commit to only include single-screen-development in the dev compose

**Comment by Matt-Wehman** [comment]
> Few small things to fix up - the `docker-compose.yml` and `README.md` should likely just be reverted to their current state in main
> 
> In addition, I couldn't get anything to show up on the screens. Please confirm that this is intended as the placeholders mentioned in the PR?
> 
> image
> I like the current UI. It's clean and reactive with snappy animations. Thanks.

The containers should be streaming, and you should be able to view the containers front end. Sometimes, the containers don't update for me until I delete my browser's cache so that may make them show up after a rebuild.

**Comment by aaron97neu** [comment]
> > Few small things to fix up - the `docker-compose.yml` and `README.md` should likely just be reverted to their current state in main
> > In addition, I couldn't get anything to show up on the screens. Please confirm that this is intended as the placeholders mentioned in the PR?
> > image
> > I like the current UI. It's clean and reactive with snappy animations. Thanks.
> 
> The containers should be streaming, so you should be able to view the front end of each vertical screen. Sometimes, the containers don't update for me until I delete my browser's cache so that may make them show up after a rebuild. What you see currently are what the placeholders were.

Interesting... This testing was on OSX, I will try a few other setups tomorrow and see if things look better. Are there anything I can do to to help debug this?

**Comment by Matt-Wehman** [comment]
> > > Few small things to fix up - the `docker-compose.yml` and `README.md` should likely just be reverted to their current state in main
> > > In addition, I couldn't get anything to show up on the screens. Please confirm that this is intended as the placeholders mentioned in the PR?
> > > image
> > > I like the current UI. It's clean and reactive with snappy animations. Thanks.
> > 
> > 
> > The containers should be streaming, so you should be able to view the front end of each vertical screen. Sometimes, the containers don't update for me until I delete my browser's cache so that may make them show up after a rebuild. What you see currently are what the placeholders were.
> 
> Interesting... This testing was on OSX, I will try a few other setups tomorrow and see if things look better. Are there anything I can do to to help debug this?
> 
> From there we can merge, possibly with future bugfixes to get it working on all platforms

I think the troubleshooting steps I would try is first completely removing the containers, rebuilding them with no cache so everything gets completely rebuilt, recomposing, deleting the browser's cache, then navigating to the localhost. Also, just as a sanity check, maybe make sure you have the most up to date code. I am attaching what the ExhibitScreen component code should look like in the latest version. I think it is a container updating issue since the green background was completely removed from the latest version so should have no way of appearing if the updated code is what is being built.

image

**Comment by Matt-Wehman** [comment]
Also, as a sanity check I did a fresh pull of my fork on the Linux machine that I got the depth sensor running on. When building from the feature branch everything works as expected. I am even able to see the depth sensor passing through the human visualization container to the dev interface! Also, if you are wondering why the screens don't have a white background is because their CSS doesn't actually declare a background color which makes them transparent. Their background is only white on their own tabs since that is the browsers default color.

image

**Comment by aaron97neu** [comment]
> Also, as a sanity check I did a fresh pull of my fork on the Linux machine that I got the depth sensor running on. When building from the feature branch everything works as expected. I am even able to see the depth sensor passing through the human visualization container to the dev interface! Also, if you are wondering why the screens don't have a white background is because their CSS doesn't actually declare a background color which makes them transparent. Their background is only white on their own tabs since that is the browsers default color.
> 
> image

Thanks for confirming. Your prediction was correct, I was looking at an outdated version of the code, which is embarrassing. Apologies. I noticed the same with the side screens, that is a pretty easy patch I can throw together. 

However, the game screen is still using the placeholder - https://github.com/conatyv8/Discovery-World-Pong-MSOE/blob/feature/single-screen-dev-ui-container/src/single-screen-development/src/Body/GameScreen.tsx#L8-L24

I assume we can snag the code used for the vert displays and then be good to go?

**Comment by Matt-Wehman** [comment]
> > Also, as a sanity check I did a fresh pull of my fork on the Linux machine that I got the depth sensor running on. When building from the feature branch everything works as expected. I am even able to see the depth sensor passing through the human visualization container to the dev interface! Also, if you are wondering why the screens don't have a white background is because their CSS doesn't actually declare a background color which makes them transparent. Their background is only white on their own tabs since that is the browsers default color.
> > image
> 
> Thanks for confirming. Your prediction was correct, I was looking at an outdated version of the code, which is embarrassing. Apologies. I noticed the same thing you did with the side screens, that is a pretty easy patch I can throw together.
> 
> However, the game screen is still using the placeholder - https://github.com/conatyv8/Discovery-World-Pong-MSOE/blob/feature/single-screen-dev-ui-container/src/single-screen-development/src/Body/GameScreen.tsx#L8-L24
> 
> I assume we can snag the code used for the vert displays and then be good to go?

Yep, the game container styling is being worked by another team member so you should still see the placeholder there.

**Review by aaron97neu — APPROVED** [review]

**Pull request merged**